### ISSUE TITLE
mi_process_init race

### ISSUE THREAD

**Issue by Novum** [microsoft/mimalloc#701]
This looks suspicious to me. If this is called from `mi_malloc_generic` from two threads at the same time isn't this completely racy?

`mi_process_init` is not guaranteed to be completed before the other thread already assumes `_mi_process_is_initialized` is set.

Maybe I'm wrong and I'm overlooking something. Or is the assumption that the main thread calls `malloc()` once before anything happens with threads?

**Comment by daanx** [comment]
Yes, it is racy. The idea is that the path through mimalloc_generic can only happen if mimalloc overrides malloc/free and it is called _before_ `main` runs (by libc or the OS loader). Here we assume that is initial code runs in a single thread. 

However, why not make it safe :-) So I pushed a fix that uses atomic initialization. Thanks!
